### PR TITLE
began fixes

### PR DESCRIPTION
**Pull request by martyn**
WIP - Still testing.

### CODE REVIEW

File: hypergan/generators/resize_conv_generator.py [hunk 1]
```diff
@@ -77,7 +77,9 @@ def create(config, gan, net, prefix="g_"):
 
     for i in range(depth):
         s = [int(x) for x in net.get_shape()]
-        layers = int(net.get_shape()[3])//depth_reduction
+        
+        #layers = int(net.get_shape()[3])//depth_reduction
+        layers = int(net.get_shape()[3])-depth_reduction
```
Comment (mikkel) [inline hypergan/generators/resize_conv_generator.py]: This should be put behind the `additive`  feature that the discriminator adds

File: hypergan/loaders/image_loader.py [hunk 2]
```diff
@@ -14,16 +14,21 @@ def build_labels(dirs):
     labels[dir.split('/')[-1]]=next_id
     next_id+=1
   return labels,next_id
-def labelled_image_tensors_from_directory(directory, batch_size, channels=3, format='jpg', width=64, height=64, crop=False, preprocess=False):
-  filenames = glob.glob(directory+"/**/*."+format)
-  labels,total_labels = build_labels(sorted(filter(os.path.isdir, glob.glob(directory+"/*"))))
+def labelled_image_tensors_from_directory(directory, batch_size, channels=3, format='jpg', width=64, height=64, crop=False, preprocess=False, filterX=None):
```
Comment (mikkel) [inline hypergan/loaders/image_loader.py]: If we don't pass in a filterX, could we just pass in different directories?

```bash
So it would be structured like:

imagedir1/a/
imagedir2/b/
```
---
Also I wonder if we should separate image alignment into another PR (or set of PRs).   There are many ways to align that need to be explored. (discoGAN, cycleGAN)

File: hypergan/trainers/joint_trainer.py [hunk 3]
```diff
@@ -105,14 +105,9 @@ def run(gan, feed_dict):
...
 
-    if(d_class_loss is not None):
-        _, g_cost,k, measure = sess.run([g_optimizer, g_loss,gan.graph.update_k], feed_dict)
-        if iteration % 100 == 0:
-            print("%2d: g cost %.2f d_loss %.2f %.6f k %.2f m" % (iteration,g_cost,  d_cost,g_k, measure))
-    else:
-        _, g_cost, g_k, measure= sess.run([g_optimizer, g_loss, gan.graph.update_k, gan.graph.measure], feed_dict)
-        if iteration % 100 == 0:
-            print("%2d: g cost %.2f d_loss %.2f  %.6f k %.2f m" % (iteration,g_cost , d_cost,g_k, measure))
+    _, g_cost, g_k, measure= sess.run([g_optimizer, g_loss, gan.graph.update_k, gan.graph.measure], feed_dict)
```
Comment (mikkel) [inline hypergan/trainers/joint_trainer.py]: This now alternates, so maybe the name `joint_trainer` is incorrect.  I'm not sure the correct name, but it trains gan.graph.update_k which is defined in `began` from `Proportional Control Theory`.

Maybe `propotional_control_trainer`?

File: hypergan/discriminators/pyramid_discriminator.py [hunk 4]
```diff
@@ -108,30 +108,24 @@ def should_ignore_layer(layer, resize):
...
+          depth = int(int(net.get_shape()[3])+depth_increase)
       else:
+          depth = int(int(net.get_shape()[3])*depth_increase)
 
-          net = conv2d(net, depth, name=prefix+'_expand_layer'+str(i), k_w=3, k_h=3, d_h=1, d_w=1, regularizer=None,gain=config.orthogonal_initializer_gain)
-
-          net = tf.nn.avg_pool(net, ksize=filter, strides=stride, padding='SAME')
+      if i ==0:
+          depth = config.first_conv_size
+      net = config.block(config, net, depth, '_layer_'+str(i)+'_')
```
Comment (martyn) [inline hypergan/discriminators/pyramid_discriminator.py]: Variable not trainable(add prefix+).

File: hypergan/loaders/image_loader.py (hunk 2 above)
Comment (martyn) [inline hypergan/loaders/image_loader.py]: Not part of this PR.  The --align flag was added, but it can be ignored for now.

File: hypergan/discriminators/pyramid_discriminator.py [hunk 5]
```diff
@@ -159,3 +153,34 @@ def should_ignore_layer(layer, resize):
     return net
 
 
+def repeating_block(config, net, depth, prefix='d_'):
```
Comment (martyn) [inline hypergan/discriminators/pyramid_discriminator.py]: move to common

File: hypergan/generators/resize_conv_generator.py (hunk 1 above)
Comment (martyn) [inline hypergan/generators/resize_conv_generator.py]: I'm not sure on that abstraction - going to skip this one.
